### PR TITLE
Fix Restoration from Seed (Old)

### PR DESCRIPTION
**Pull request by nielsandriesse**
#221 with changes

### CODE REVIEW

File: Signal/src/Loki/View Controllers/RestoreVC.swift [hunk 1]
```diff
@@ -164,6 +164,13 @@ final class RestoreVC : BaseVC {
             OWSPrimaryStorage.shared().setRestorationTime(Date().timeIntervalSince1970)
             UserDefaults.standard[.hasViewedSeed] = true
             mnemonicTextField.resignFirstResponder()
+            let _ = LokiFileServerAPI.getDeviceLinks(associatedWith: (keyPair.hexEncodedPublicKey)).done2{ deviceLinks in
+                //Send friend requests to linked devices when restore from seed
+                let selfDeviceLinks = deviceLinks.filter{$0.master.hexEncodedPublicKey == getUserHexEncodedPublicKey()}
+                for devicelink in selfDeviceLinks {
+                    MultiDeviceProtocol.sendSessionResetRequestToLinkedDevice(to: devicelink.slave.hexEncodedPublicKey)
```
Comment (Mikunj) [inline Signal/src/Loki/View Controllers/RestoreVC.swift:171]: is `SessionResetRequest` just a `SessionRequest`?

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 2]
```diff
@@ -1062,7 +1066,7 @@ - (void)updateSessionRestoreBanner {
     BOOL shouldDetachBanner = !isContactThread;
     if (isContactThread) {
         TSContactThread *thread = (TSContactThread *)self.thread;
-        if (thread.sessionRestoreDevices.count > 0) {
+        if (thread.sessionRestoreDevices.count > 0 && thread.sessionResetStatus == LKSessionResetStatusNone) {
```
Comment (Mikunj) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m:1069]: A change we just made recently on desktop is that we no longer ask the user if sessions need to be reset. Basically on restore if you can't decrypt or you get `no record for session` error then you send out a `SessionRequest` if you haven't done it before.

File: SignalServiceKit/src/Loki/Protocol/Session Management/LokiSessionResetImplementation.swift [hunk 3]
```diff
@@ -44,19 +45,29 @@ public class LokiSessionResetImplementation : NSObject, SessionResetProtocol {
...
         guard !recipientID.isEmpty else { return }
+        let masterHexEncodedPublicKey = storage.getMasterHexEncodedPublicKey(for: recipientID, in: transaction) ?? recipientID
         guard let thread = TSContactThread.getWithContactId(recipientID, transaction: transaction) else {
             Logger.debug("[Loki] A new session was adopted but the thread couldn't be found for: \(recipientID).")
             return
         }
         // If the current user initiated the reset then send back an ephemeral message to acknowledge the completion of the session reset
-        if thread.sessionResetStatus == .initiated {
-            SessionManagementProtocol.sendSessionEstablishedMessage(to: recipientID, in: transaction)
+        if let masterThread = TSContactThread.getWithContactId(masterHexEncodedPublicKey, transaction: transaction), masterThread.sessionResetStatus == .initiated {
```
Comment (Mikunj) [inline SignalServiceKit/src/Loki/Protocol/Session Management/LokiSessionResetImplementation.swift:54]: What about cases where you also want to reset the session with a secondary device? This is probably the direction we're heading in the future where triggering a session reset should send out the message to all linked devices

**Comment by nielsandriesse** [comment]
This PR has been superseded

**Pull request closed without merging**